lib/masdb/data_server.ex

```diff
@@ -0,0 +1,48 @@
+defmodule Masdb.Data.Server do
+  use GenServer
+  alias Masdb.Data
+  alias Masdb.Register
+  alias Masdb.Timestamp
+
+  def start_link(register \\ Register.Server, name \\ __MODULE__) do
+    GenServer.start_link(__MODULE__,
+      {
+        register,
+        %Data.Map{
+          node_id: inspect(Node.self),
+          last_update_time: Timestamp.get_timestamp()
+        }},
+      name: name
+    )
+  end
+
+  def init(state) do
+   {:ok, state}
+  end
+
+  def insert(schema_name, values, name \\ __MODULE__) do
+    GenServer.call(name, {:insert, schema_name, values})
+  end
+
+  # private
+  def handle_call(params, from, {register, %Data.Map{}} = state) do
+    # since this is blocking, it could cause issues
+    if Register.Server.is_synced?(register) do
```

Comment:
Since we are not keeping (as I just saw below) the schema into the datamap... You can ignore the previous comment.